server/src/editor.js

```diff
@@ -0,0 +1,140 @@
+const React = require("react");
+const sendEvent = require("./browser-send-event.js");
+
+let pos, context, editor, highlightContext, highlighter, drawContext;
+
+exports.Editor = class Editor extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tool: 'none',
+      highlighter: 'inactive',
+      pen: 'inactive',
```

Comment:
The highlighter and the pen states are more for the state of their respective icons than anything else, but I'm sure I can figure out a way to have only one tool state. @ianb